packages/query-core/src/types.ts

```diff
@@ -319,7 +319,12 @@ export interface QueryObserverOptions<
    */
   placeholderData?:
     | NonFunctionGuard<TQueryData>
-    | PlaceholderDataFunction<NonFunctionGuard<TQueryData>>
+    | PlaceholderDataFunction<
+        NonFunctionGuard<TQueryData>,
```

Comment:
I'm not sure if this should be `TQueryFnData`?

Reply:
no I think this is fine. The difference is only for InfiniteQueries, where one is wrapped in `InfiniteData`. But maybe you can add a couple of type tests?

Reply:
Sure, I added some tests. The last one would have also passed without my changes. Let me know if it needs more work.